jujugui/static/gui/src/app/utils/environment-change-set.js

```diff
@@ -572,33 +538,35 @@ YUI.add('environment-change-set', function(Y) {
       Receives all the parameters received by the models "addCharm" method
       with the exception of the ECS options object.
 
-      @method _lazyAddCharm
+      @method lazyAddCharm
       @param {Array} args The arguments to add the charm with.
+      @param {Object} options The ECS options.
     */
-    _lazyAddCharm: function(args) {
+    lazyAddCharm: function(args, options) {
       const existing = Object.keys(this.changeSet).some(key => {
         return this.changeSet[key].command.args[0] === args[0];
       });
       // If there is an existing record for this charm then don't add another.
       if (existing) {
         return;
       }
-      var command = {
-        method: '_addCharm',
-        args: this._getArgs(args),
-        options: args[3]
-      };
+      const command = {method: '_addCharm', args: args, options: options};
```

Comment:
Since you are doing edits for these commands, you can use object shorthand for the repeated properties (and the calls below)
```JavaScript
const command = {method: '_addCharm', args, options};
```